contracts/Aavegotchi/facets/ItemsFacet.sol

```diff
@@ -182,38 +190,76 @@ contract ItemsFacet is Modifiers {
     ///@dev A wearable cannot be equipped in the wrong slot
     ///@param _tokenId The identifier of the aavegotchi to make changes to
     ///@param _wearablesToEquip An array containing the identifiers of the wearables to equip
-    function equipWearables(uint256 _tokenId, uint16[EQUIPPED_WEARABLE_SLOTS] calldata _wearablesToEquip)
+    function equipWearables(
+        uint256 _tokenId,
+        uint16[EQUIPPED_WEARABLE_SLOTS] calldata _wearablesToEquip
+    )
         external
+    {
+        uint256[EQUIPPED_WEARABLE_SLOTS] memory _commitmentIds;
+        _equipWearables(_tokenId, _wearablesToEquip, _commitmentIds);
+    }
+
+    ///@notice Allow the owner of a claimed aavegotchi to equip/unequip wearables to his aavegotchi
+    ///@dev Only valid for claimed aavegotchis
+    ///@dev A zero value will unequip that slot and a non-zero value will equip that slot with the wearable whose identifier is provided
+    ///@dev A wearable cannot be equipped in the wrong slot
+    ///@param _tokenId The identifier of the aavegotchi to make changes to
+    ///@param _wearablesToEquip An array containing the identifiers of the wearables to equip
+    ///@param _commitmentIds An array containing the identifiers of the deposited wearables to equip
+    function equipDelegatedWearables(
+        uint256 _tokenId,
+        uint16[EQUIPPED_WEARABLE_SLOTS] calldata _wearablesToEquip,
+        uint256[EQUIPPED_WEARABLE_SLOTS] calldata _commitmentIds
+    )
+        external
+    {
+        
+        _equipWearables(_tokenId, _wearablesToEquip, _commitmentIds);
+    }
+
+
+    function _equipWearables(
+        uint256 _tokenId,
+        uint16[EQUIPPED_WEARABLE_SLOTS] calldata _wearablesToEquip,
+        uint256[EQUIPPED_WEARABLE_SLOTS] memory _commitmentIdsToEquip
+    )
+        internal
         onlyAavegotchiOwner(_tokenId)
         onlyUnlocked(_tokenId)
     {
         Aavegotchi storage aavegotchi = s.aavegotchis[_tokenId];
         require(aavegotchi.status == LibAavegotchi.STATUS_AAVEGOTCHI, "LibAavegotchi: Only valid for AG");
-        emit EquipWearables(_tokenId, aavegotchi.equippedWearables, _wearablesToEquip);
-
-        address sender = LibMeta.msgSender();
+        emit EquipWearables(_tokenId, s.aavegotchis[_tokenId].equippedWearables, _wearablesToEquip);
+        emit EquipDelegatedWearables(_tokenId, s.gotchiEquippedItemsInfo[_tokenId].equippedCommitmentIds, _commitmentIdsToEquip);
+        GotchiEquippedCommitmentsInfo storage _gotchiInfo = s.gotchiEquippedItemsInfo[_tokenId];
```

Comment:
Commitments is a strange word, is there a reason you're using this word to represent the concept of the delegated wearables?

Reply:
In the context of ERC-7589, there are two ways of delegating assets:
* You can either deposit the tokens on an intermediary contract (this is the Wearables case).
* Or you can freeze the user's balance (prevent it from being transferred).
In the beginning, we used the word "deposit" and "withdraw" to represent these events, but because they didn't make sense for use cases when tokens are frozen, we decided to move to a more neutral word. "Commitment" was the word we chose, in the sense that the user is "committing tokens". But I can understand how that can be confusing on the context of wearables.

That being said, we can change from "commitment" to a friendlier term. How about "deposit"? We can change `commitmentId` to `depositId` for example